woocommerce/woocommerce-ios
# Readonly Product Variations: Variations & Product Details UI

Fixes #1469 
Fixes #1470 

Summary: Add a Product variants info row to the Product Details screen, and navigate to a new Product Variations paginated list screen on tapping the variants info row.

## Changes
- Added a new feature flag `readonlyProductVariants`
- On Product Details screen, conditionally showed the Product variants info (Product attributes with their number of options) based on the feature flag and whether the Product has non-empty variations
- Created `ProductVariationsViewController` that displays a paginated table view of Product Variations with the data from `ProductVariationAction.synchronizeProductVariations`
  - Added a `ProductsTabProductViewModel` extension to set up the view model with a `ProductVariation`
- Navigated to `ProductVariationsViewController` when the user taps on the Product variants info row on Product Details screen

## Testing
Prerequisites: have two types of Products - simple and variable Products with variations
- Launch the app
- Go to the Products tab
- Tap into a simple Product --> should see the Price & Inventory section on the Product Details screen
- Go back to the Products tab
- Tap into a variable Product --> should see the variants info below the reviews row with each attribute and the number of options, and should **not** see the Price & Inventory section on the Product Details screen
- Tap the variants info row --> should see a list of variations with the price or visibility


## Example screenshots

Product has no variations | Product has variations | Variations UI
-- | -- | --
![Simulator Screen Shot - iPhone 8 - 2019-11-22 at 09 17 09](https://user-images.githubusercontent.com/1945542/69390549-8cb92780-0d0a-11ea-885a-a3246437af9f.png) | ![Simulator Screen Shot - iPhone 8 - 2019-11-22 at 09 08 32](https://user-images.githubusercontent.com/1945542/69390547-8c209100-0d0a-11ea-8a6f-142d4db3c486.png) | ![Simulator Screen Shot - iPhone 8 - 2019-11-22 at 09 08 38](https://user-images.githubusercontent.com/1945542/69390548-8cb92780-0d0a-11ea-8c28-23948904fa71.png)

Update release notes:
